diff --git a/drivers/usb/serial/ftdi_sio.c b/drivers/usb/serial/ftdi_sio.c
index 05e28a5ce..c5f653e4f 100644
--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -2311,6 +2311,12 @@ static int ftdi_8u2232c_probe(struct usb_serial *serial)
 		 !strcmp(udev->product, "SNAP Connect E10")))
 		return ftdi_jtag_probe(serial);
 
+	if (udev->product &&
+		(!strcmp(udev->product, "SDS TypeX USB3 Parallel"))) {
+		dev_info(&udev->dev, "Ignoring Parallel interface for SDS TypeX\n");
+		return -ENODEV;
+	}
+
 	return 0;
 }
 
